hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -76,6 +76,10 @@ public BaseCommitActionExecutor(HoodieEngineContext context, HoodieWriteConfig c
     // TODO : Remove this once we refactor and move out autoCommit method from here, since the TxnManager is held in {@link AbstractHoodieWriteClient}.
     this.txnManager = new TransactionManager(config, table.getMetaClient().getFs());
     this.lastCompletedTxn = TransactionUtils.getLastCompletedTxnInstantAndMetadata(table.getMetaClient());
+    if (table.getStorageLayout().operationConstraint(operationType)) {
```

Comment:
Thanks for doing this. the storage layout abstraction is something I wish we added before. but when done with real use-cases, makes it lot easier.